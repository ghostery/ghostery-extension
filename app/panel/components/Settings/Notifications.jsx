/**
 * Notifications Settings Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
/**
 * @class Implement Notification subview as a React component.
 * The view opens from the left-side menu of the main
 * Settings view.
 * @memberOf SettingsComponents
 */
const Notifications = ({ settingsData, toggleCheckbox }) => (
	<div className="s-tabs-panel">
		<div className="row">
			<div className="columns">
				<h3>{ t('settings_notifications') }</h3>
				<h5>{ t('settings_notify_me') }</h5>
				<div className="s-tooltip-down" data-g-tooltip={t('settings_notify_me_tooltip')}>
					<img src="../../app/images/panel/icon-information-tooltip.svg" className="s-question" />
				</div>
				<div className="s-option-group">
					<div className="s-square-checkbox">
						<input type="checkbox" id="settings-announcements" name="show_cmp" defaultChecked={settingsData.show_cmp} onClick={toggleCheckbox} />
						<label id="settings-announcements-label" htmlFor="settings-announcements">{ t('settings_announcements') }</label>
					</div>
				</div>
				<div className="s-option-group">
					<div className="s-square-checkbox">
						<input type="checkbox" id="settings-new-features" name="notify_upgrade_updates" defaultChecked={settingsData.notify_upgrade_updates} onClick={toggleCheckbox} />
						<label id="settings-new-features-label" htmlFor="settings-new-features">{ t('settings_new_features') }</label>
					</div>
				</div>
				<div className="s-option-group">
					<div className="s-square-checkbox">
						<input type="checkbox" id="settings-new-promotions" name="notify_promotions" defaultChecked={settingsData.notify_promotions} onClick={toggleCheckbox} />
						<label id="settings-new-promotions-label" htmlFor="settings-new-promotions">{ t('settings_new_promotions') }</label>
					</div>
				</div>
				<div className="s-option-group">
					<div className="s-square-checkbox">
						<input type="checkbox" id="settings-new-trackers" name="notify_library_updates" defaultChecked={settingsData.notify_library_updates} onClick={toggleCheckbox} />
						<label id="settings-new-trackers-label" htmlFor="settings-new-trackers">{ t('settings_new_trackers') }</label>
					</div>
				</div>
				<div className="s-option-group">
					<div className="s-square-checkbox">
						<input type="checkbox" id="settings-show-reload-banner" name="reload_banner_status" defaultChecked={settingsData.reload_banner_status} onClick={toggleCheckbox} />
						<label id="settings-show-reload-banner-label" htmlFor="settings-show-reload-banner">{ t('settings_show_reload_banner') }</label>
					</div>
				</div>
				<div className="s-option-group">
					<div className="s-square-checkbox">
						<input type="checkbox" id="settings-show-trackers-banner" name="trackers_banner_status" defaultChecked={settingsData.trackers_banner_status} onClick={toggleCheckbox} />
						<label id="settings-show-trackers-banner-label" htmlFor="settings-show-trackers-banner">{ t('settings_show_trackers_banner') }</label>
					</div>
				</div>
				<div className="s-option-group">
					<div className="s-square-checkbox">
						<input type="checkbox" id="settings-show-count-badge" name="show_badge" defaultChecked={settingsData.show_badge} onClick={toggleCheckbox} />
						<label id="settings-show-count-badge-label" htmlFor="settings-show-count-badge">{ t('settings_show_tracker_count_badge') }</label>
					</div>
				</div>
			</div>
		</div>
	</div>
);

Notifications.propTypes = {
	toggleCheckbox: PropTypes.func.isRequired,
	settingsData: PropTypes.shape({
		show_cmp: PropTypes.bool.isRequired,
		notify_upgrade_updates: PropTypes.bool.isRequired,
		notify_promotions: PropTypes.bool.isRequired,
		notify_library_updates: PropTypes.bool.isRequired,
		reload_banner_status: PropTypes.bool,
		trackers_banner_status: PropTypes.bool,
		show_badge: PropTypes.bool.isRequired,
	}).isRequired,
};

export default Notifications;
