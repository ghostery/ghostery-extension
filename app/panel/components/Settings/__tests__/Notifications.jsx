/**
 * Notifications Settings Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Notifications from '../Notifications';

describe('app/panel/Settings/Notifications.jsx', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('Notifications is rendered correctly with falsy props', () => {
			const settingsData = {
				show_cmp: false,
				notify_upgrade_updates: false,
				notify_promotions: false,
				notify_library_updates: false,
				reload_banner_status: false,
				trackers_banner_status: false,
				show_badge: false,
			};

			const component = renderer.create(
				<Notifications
					settingsData={settingsData}
					toggleCheckbox={() => {}}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('Notifications is rendered correctly with truthy props', () => {
			const settingsData = {
				show_cmp: true,
				notify_upgrade_updates: true,
				notify_promotions: true,
				notify_library_updates: true,
				reload_banner_status: true,
				trackers_banner_status: true,
				show_badge: true,
			};

			const component = renderer.create(
				<Notifications
					settingsData={settingsData}
					toggleCheckbox={() => {}}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('Notifications functions correctly', () => {
			const settingsData = {
				show_cmp: false,
				notify_upgrade_updates: false,
				notify_promotions: false,
				notify_library_updates: false,
				reload_banner_status: false,
				trackers_banner_status: false,
				show_badge: false,
			};
			const toggleCheckbox = jest.fn();

			const component = mount(
				<Notifications
					settingsData={settingsData}
					toggleCheckbox={toggleCheckbox}
				/>
			);

			expect(toggleCheckbox.mock.calls.length).toBe(0);
			component.find('#settings-announcements').simulate('click');
			component.find('#settings-new-features').simulate('click');
			component.find('#settings-new-promotions').simulate('click');
			component.find('#settings-new-trackers').simulate('click');
			component.find('#settings-show-reload-banner').simulate('click');
			component.find('#settings-show-trackers-banner').simulate('click');
			component.find('#settings-show-count-badge').simulate('click');
			expect(toggleCheckbox.mock.calls.length).toBe(7);
		});
	});
});
