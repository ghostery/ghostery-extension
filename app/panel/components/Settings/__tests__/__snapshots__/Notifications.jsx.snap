// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/panel/Settings/Notifications.jsx Snapshot tests with react-test-renderer Notifications is rendered correctly with falsy props 1`] = `
<div
  className="s-tabs-panel"
>
  <div
    className="row"
  >
    <div
      className="columns"
    >
      <h3>
        settings_notifications
      </h3>
      <h5>
        settings_notify_me
      </h5>
      <div
        className="s-tooltip-down"
        data-g-tooltip="settings_notify_me_tooltip"
      >
        <img
          className="s-question"
          src="../../app/images/panel/icon-information-tooltip.svg"
        />
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={false}
            id="settings-announcements"
            name="show_cmp"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-announcements"
            id="settings-announcements-label"
          >
            settings_announcements
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={false}
            id="settings-new-features"
            name="notify_upgrade_updates"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-new-features"
            id="settings-new-features-label"
          >
            settings_new_features
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={false}
            id="settings-new-promotions"
            name="notify_promotions"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-new-promotions"
            id="settings-new-promotions-label"
          >
            settings_new_promotions
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={false}
            id="settings-new-trackers"
            name="notify_library_updates"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-new-trackers"
            id="settings-new-trackers-label"
          >
            settings_new_trackers
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={false}
            id="settings-show-reload-banner"
            name="reload_banner_status"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-show-reload-banner"
            id="settings-show-reload-banner-label"
          >
            settings_show_reload_banner
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={false}
            id="settings-show-trackers-banner"
            name="trackers_banner_status"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-show-trackers-banner"
            id="settings-show-trackers-banner-label"
          >
            settings_show_trackers_banner
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={false}
            id="settings-show-count-badge"
            name="show_badge"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-show-count-badge"
            id="settings-show-count-badge-label"
          >
            settings_show_tracker_count_badge
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`app/panel/Settings/Notifications.jsx Snapshot tests with react-test-renderer Notifications is rendered correctly with truthy props 1`] = `
<div
  className="s-tabs-panel"
>
  <div
    className="row"
  >
    <div
      className="columns"
    >
      <h3>
        settings_notifications
      </h3>
      <h5>
        settings_notify_me
      </h5>
      <div
        className="s-tooltip-down"
        data-g-tooltip="settings_notify_me_tooltip"
      >
        <img
          className="s-question"
          src="../../app/images/panel/icon-information-tooltip.svg"
        />
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={true}
            id="settings-announcements"
            name="show_cmp"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-announcements"
            id="settings-announcements-label"
          >
            settings_announcements
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={true}
            id="settings-new-features"
            name="notify_upgrade_updates"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-new-features"
            id="settings-new-features-label"
          >
            settings_new_features
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={true}
            id="settings-new-promotions"
            name="notify_promotions"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-new-promotions"
            id="settings-new-promotions-label"
          >
            settings_new_promotions
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={true}
            id="settings-new-trackers"
            name="notify_library_updates"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-new-trackers"
            id="settings-new-trackers-label"
          >
            settings_new_trackers
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={true}
            id="settings-show-reload-banner"
            name="reload_banner_status"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-show-reload-banner"
            id="settings-show-reload-banner-label"
          >
            settings_show_reload_banner
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={true}
            id="settings-show-trackers-banner"
            name="trackers_banner_status"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-show-trackers-banner"
            id="settings-show-trackers-banner-label"
          >
            settings_show_trackers_banner
          </label>
        </div>
      </div>
      <div
        className="s-option-group"
      >
        <div
          className="s-square-checkbox"
        >
          <input
            defaultChecked={true}
            id="settings-show-count-badge"
            name="show_badge"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="settings-show-count-badge"
            id="settings-show-count-badge-label"
          >
            settings_show_tracker_count_badge
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
